src/github/dependabot.ts

```diff
@@ -43,6 +44,60 @@ export interface DependabotOptions {
    * List of labels to apply to the created PR's.
    */
   readonly labels?: string[];
+
+  /**
+   * Map of dependabot registries to use
+   * @default undefined
+   */
+  readonly registries?: { [name: string]: DependabotRegistry };
+}
+
+/**
+ * Use to add private registry support for dependabot
+ * https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#configuration-options-for-private-registries
+ */
+export interface DependabotRegistry {
+  /**
+   * Registry type e.g. 'npm-registry' or 'docker-registry'
+   */
+  readonly type: string;
+
+  /**
+   * Url for the registry e.g. 'https://npm.pkg.github.com' or 'registry.hub.docker.com'
+   */
+  readonly url: string;
+
+  /**
+   * The username that Dependabot uses to access the registry
+   * @default undefined
```

Comment:
In such cases we prefer to specify the default behavior (i.e. what `undefined` means):

```suggestion
   * @default - do not authenticate
```